comixed-services/src/main/java/org/comixed/service/library/PageCacheService.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * ComiXed - A digital comic book library management application.
+ * Copyright (C) 2019, The ComiXed Project
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses>
+ */
+
+package org.comixed.service.library;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+public class PageCacheService {
+  protected final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+  @Value("${comixed.images.cache.location}")
+  private String cacheDirectory;
+
+  public byte[] findByHash(final String hash) {
+    this.logger.debug("Searching for cached image: hash={}", hash);
+
+    final File file = this.getFileForHash(hash);
+    if (file.exists() && !file.isDirectory()) {
+      this.logger.debug("Loading cached image content: {} bytes", file.length());
+      try {
+        return IOUtils.readFully(new FileInputStream(file), (int) file.length());
+      } catch (Exception error) {
+        this.logger.error("Failed to load cached image", error);
+      }
```

Comment:
Add a finally close of the of the open file so that in case of exception we don't get leaks

Reply:
Not as a finally, which require another level of try/catch coding. But I refactored it to hold a reference to the input stream so it can be explicitly closed.